File: C++/concatenation-of-array.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> result(nums.size() * 2);
        
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            result[i] = result[i+n] = nums[i];
        }
        
        return result;
    }
};

File: JavaScript/concatenation-of-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getConcatenation = function (nums) {
    const n = nums.length
    const result = Array(n)
    for (let i = 0; i < n; i++) {
        result[i] = result[i + n] = nums[i]
    }
    return result
};